src/processing/server/board/ServerCommunication.java

```diff
@@ -0,0 +1,149 @@
+package processing.server.board;
+
+import processing.ClientBoardState;
+
+import java.io.IOException;
+
+import infrastructure.validation.logger.LogLevel;
+import infrastructure.validation.logger.ModuleID;
+import processing.*;
+import processing.boardobject.*;
+
+/**
+ * This class implements IServerCommunication interface, hencke defines all
+ * the functions required by the client for communication with the BoardServer
+ * 
+ * @author Himanshu Jain
+ * @reviewer Ahmed Zaheer Dadarkar
+ *
+ */
+
+public class ServerCommunication implements IServerCommunication{
+	
+	// variable to store the server address with ip and port
+	String serverAddress;
+	
+	/**
+	 * Constructor to set the server address as the object of this
+	 * class is created to use this server address for further 
+	 * communications in this class.
+	 */
+	public ServerCommunication() {
+		
+		/**
+		 * setting serverAddress as the full server address which we
+		 * need to pass the destination in the send function of networking
+		 * module.
+		 */
+		serverAddress = ClientBoardState.serverIp.toString()
+					  + ":"
+					  + ClientBoardState.serverPort.toString();
+	}
+
+	/**
+	 * This function will be called in the start when the client will receive
+	 * the port number of the Board Server to get the persistence Board State 
+	 * if there from the server.
+	 */
+	public void getBoardState() {
+		
+		/**
+		 * While sending BoardState request to the server, as it is the starting
+		 * communication of the client with the server, we will pass the client
+		 * details in the message so that server will save this information for
+		 * further use.
+		 * 
+		 * We are sending username and the client full address
+		 */
+		String message = ClientBoardState.userId.toString()
+					   + ":"
+					   + ClientBoardState.userIP.toString()
+					   + ":"
+					   + ClientBoardState.userPort.toString();
```

Comment:
Can we use JSON instead? It will give us more flexibility later on, to add more fields if required.

Reply:
Okay